Logic/igtlioConnector.cxx

```diff
@@ -408,11 +405,17 @@ int igtlioConnector::ReceiveController()
 
   while (!this->ServerStopFlag)
   {
-  igtlioLockGuard<vtkMutexLock> lock(this->Mutex);
-  for (std::vector<Client>::iterator clientIt = this->Sockets.begin(); clientIt != this->Sockets.end(); ++clientIt)
+    std::vector<int> clientIds = this->GetClientIds();
+    for (std::vector<int>::iterator clientIdIt = clientIds.begin(); clientIdIt != clientIds.end(); ++clientIdIt)
     {
+      Client client = this->GetClient(*clientIdIt);
+      if (client.ID == -1)
+      {
+        continue;
```

Comment:
this change will break if the vector changes during iteration, no? if the vector is shortened to a size smaller than the current pointer of clientIdIt line 409 will throw an out-of bounds exception, right?

if you are going to do it this way, do it by
```c++
for(std::vector<int>::sizse_type i = 0; i < clientIds.size(); ++i)
{
...
}
```

Reply:
clientIds is a copy

Reply:
ah, yes, right

we talked a long time ago about return types possibly being optimized by the compiler, is there a directive to say always return a copy?